tests/run-make/libtest-thread-limit/test.rs

```diff
@@ -10,7 +10,11 @@ fn spawn_thread_would_block() {
     THREAD_ID.set(thread::current().id()).unwrap();
 }
 
+// HACK(Oneirical): Tests are run in alphabetical order, and the second test is dependent on the
+// first to set THREAD_ID. Do not rename the tests in such a way that `test_run_in_same_thread`
+// would run before `spawn_thread_would_block`.
```

Comment:
Discussion: I'm not a massive fan of this, I would not try to rely on libtest execution order **unless** this is a property that T-libs desires.

Reply:
cc @rust-lang/libs: Hi T-libs, for libtest, is this a **documented**/**guaranteed** property that tests are exercised in alphabetical order under thread exhaustion scenario, do we want to check this?

Reply:
I don't think that there's anything special about the test order under thread exhaustion. The default order is documented to be alphabetical (https://doc.rust-lang.org/rustc/tests/index.html#--shuffle), so I think that it is fine to depend on that.